Dp-202-shop/server/src/common/dtos/search.params.dto.ts

```diff
@@ -0,0 +1,17 @@
+import * as Joi from 'joi';
+import {  ValidatedRequestSchema, createValidator } from 'express-joi-validation';
+
+
+export const validator = createValidator();
+
+export interface ISearchParams extends ValidatedRequestSchema {
+    categories?: string;
+    manufactures?: string;
+    products?: string
+}
+
+export const searchParamsDto = Joi.object().keys({
+    categories: Joi.string().pattern(/^[0-9,]+$/).allow(''),
+    manufactures: Joi.string().allow(''),
+    products: Joi.string().allow('')
+}).unknown(true);
```

Comment:
why do we allow('') everywhere?

Reply:
> why do we allow('') everywhere?

Because I wanted /search to Wark even without parameters